connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -329,6 +334,48 @@ public Set<String> createTopics(NewTopic... topics) {
         return createOrFindTopics(topics).createdTopics();
     }
 
+    /**
+     * Implements a retry logic around creating topic(s) in case it'd fail due to InvalidReplicationFactorException
+     *
+     * @param topicDescription
+     * @param config
+     * @param time
+     * @return the same as {@link TopicAdmin#createTopics(NewTopic...)}
+     */
+    public Set<String> createTopicsWithRetry(NewTopic topicDescription, WorkerConfig config, Time time) {
```

Comment:
I would avoid passing a full WorkerConfig here, and try to minimalize the amount of data passed here by only propagating the timeout and the backoff. That simplifies testing, too.

Reply:
This was exactly what I don't like in my change :D 
I was trying to avoid duplicating the code for this instanceof check. I'll find another place.